### PR TITLE
Fix `Button` minimum size when `disabled` is toggled.

### LINKED ISSUE #95580: Cutting text in button StyleBox with Context Margins
### Tested versions

4.3 stable

### System information

Windows 10 Godot 4.3 stable Forward+

### Issue description

The last letters are cut off for different styleboxes in the button

issue

Disabled (Empty)             |  Normal
:-------------------------:|:-------------------------:
  |  

### Steps to reproduce

> Empty project, just create a scene with a container
> In container create button with both central alignment.

When the button is disabled and the text is changed when it's disabled, when set 'Disabled ' to false, the last characters are cut

At runtime text is cut off simply by setting 'Disabled' off from script (`button.disabled = false`) when the button was initially disabled in the scene

### Minimal reproduction project (MRP)

[issue_scene.zip](https://github.com/user-attachments/files/16628057/issue.zip)

### PR DESCRIPTION
**Pull request by WhalesState**
* Fixes: https://github.com/godotengine/godot/issues/95580

### CODE REVIEW

**Comment by KoBeWi** [comment]
Updating minimum size in `NOTIFICATION_DRAW` means that the Button's size is undefined until it's drawn for the first time. This can cause various sizing problems with containers.

**Comment by WhalesState** [comment]
> Updating minimum size in `NOTIFICATION_DRAW` means that the Button's size is undefined until it's drawn for the first time. This can cause various sizing problems with containers.

It's undifined

> Updating minimum size in `NOTIFICATION_DRAW` means that the Button's size is undefined until it's drawn for the first time. This can cause various sizing problems with containers.

We can use `_shape()` inside `get_minimum_size()` similar to how `Label` do, but I couldn't find any issues in editor since everything was fine after this change.

The reason why i have done this is because I have compared both `Label` and `Button` to check what happens when the `font_oversampling` changed, the `Label` is taking care of everything inside the draw notfication.

File: scene/gui/button.cpp [hunk 1]
```diff
@@ -187,29 +187,22 @@ void Button::_notification(int p_what) {
...
 			if (autowrap_mode != TextServer::AUTOWRAP_OFF) {
-				_shape();
-
-				update_minimum_size();
 				queue_redraw();
 			}
 		} break;
 
 		case NOTIFICATION_DRAW: {
+			_shape();
```
[the remark this answers is not in the document]
Comment (bruvzg) [reply scene/gui/button.cpp:204]: It probably should do something similar to `Label`, and check if shaped text was invalidated by font change (with `shaped_text_is_ready`), and only then call `_shape`.

https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/scene/gui/label.cpp#L407-L417

[the remark this answers is not in the document]
Comment (bruvzg) [reply scene/gui/button.cpp:204]: It probably should do something similar to `Label`, and check if shaped text was invalidated by font change (with `shaped_text_is_ready`), and only then call `_shape`.

https://github.com/godotengine/godot/blob/db66bd35af704fe0d83ba9348b8c50a48e51b2ba/scene/gui/label.cpp#L407-L417

**Review by bruvzg — CHANGES_REQUESTED** [review]

File: scene/gui/button.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (bruvzg) [reply scene/gui/button.cpp:204]: Actually, I completely forgot I did #95511, which do exactly this. Will re-check it tomorrow.

[the remark this answers is not in the document]
Comment (bruvzg) [reply scene/gui/button.cpp:204]: > Will re-check it tomorrow.

Fixes https://github.com/godotengine/godot/issues/97417 but not https://github.com/godotengine/godot/issues/95580

I guess for #95580 we can add override for `set_disabled` to reshape it.

[the remark this answers is not in the document]
Comment (WhalesState) [reply scene/gui/button.cpp:204]: I stumbled upon this draft PR and wanted to spark a discussion. To ensure correct updates when `queue_redraw` is called, perhaps we should follow the approach used by Label or explore alternative solutions.

[the remark this answers is not in the document]
Comment (WhalesState) [reply scene/gui/button.cpp:204]: I will draft this PR, until you check (https://github.com/godotengine/godot/issues/97417).

**Comment by WhalesState** [comment]
I just have fixed the minimum size when disabled is toggled, since you already have fixed the other issue.

**Review by bruvzg — APPROVED** [review]
Tested with https://github.com/godotengine/godot/issues/95580 scene, seems to be working fine.

**Comment by KoBeWi** [comment]
Why was this closed?

**Comment by markdibarry** [comment]
It seems they moved to a different fork project of Godot and closed all of their PRs, so I assume they're up for grabs for anyone who wants to reopen them as a new PR.

**Comment by KoBeWi** [comment]
Except not all were closed https://github.com/godotengine/godot/pulls/whalesstate 🤔

Some of the closed PRs were already approved and did not require further action. You can still restore the branch and reopen them.

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**